compiler/ballerina-lang/src/main/resources/compiler.properties

```diff
@@ -1628,3 +1628,15 @@ error.invalid.isolated.qualifier.on.module.no.init.var.decl=\
 
 error.binding.pattern.not.yet.supported.in.module.var.decl=\
   ''{0}'' binding pattern not yet supported in module variable declaration
+
+error.invalid.read.only.class.inclusion.in.object.type.descriptor=\
+  object type inclusion cannot be used with a ''readonly class'' in an object type descriptor
+
+error.invalid.inclusion.with.mismatched.qualifiers=\
+  invalid object type inclusion: missing ''{0}'' qualifier(s) in the referencing object
+
+error.invalid.read.only.class.inclusion.in.non.read.only.class=\
```

Comment:
In `object constructor` type-reference is implemented as a type inclusion. This may be confusing in some scenarios if the error refers to it as "type inclusion".

Reply:
This error will only be logged for user-defined classes, not for object constructor expressions.

In an object constructor expression, if the type reference is a `readonly class`, the rules that apply are somewhat the same as those within a `readonly class`.

e.g.,

The fields become final and the types becomes the types with their read-only bit set.
```ballerina
readonly class Foo {
    int i;

    function init(int i) {
        self.i = i;
    }
}

public function main() {
    object { } x =  object Foo { 
        int i = 1;
        map<string> m = mp; // `mp` should be a subtype of `map<string> & readonly`
    }; 
}
```

If the type reference is a read-only class, the object constructor's fields have to be `readonly` compatible.
```ballerina
readonly class Foo {
    int i;

    function init(int i) {
        self.i = i;
    }
}

public function main() {
    object { } x =  object Foo { 
        int i = 1;
        stream<int> str; // can never be a subtype of `readonly`

        function init() {
            int[] arr = [1, 2, 3, 4];
            self.str = arr.toStream();
        }
    }; 
}
```
The error logged here will be "invalid field in an object constructor expression with a 'readonly' reference: 'stream<int>' can never be 'readonly'"

Spec discussion - https://github.com/ballerina-platform/ballerina-spec/issues/649

Reply:
Sorry for not being clear. Check this example.
```
isolated class Foo {
    int i;

    function init(int i) {
        self.i = i;
    }
}

public function main() {
    object { } x =  object Foo { 
        int i = 1;
    }; 
}
```

```LogTalk
ERROR [t14.bal:(10:28,10:31)] invalid object type inclusion: missing 'isolated' qualifier(s) in the referencing object
```

Reply:
Introduced a separate error for object constructor expressions in d057072.